Kernel/Tasks/Process.cpp

```diff
@@ -293,16 +293,33 @@ void Process::unprotect_data()
     });
 }
 
-ErrorOr<Process::ProcessAndFirstThread> Process::create_with_forked_name(UserID uid, GroupID gid, ProcessID ppid, bool is_kernel_process, NonnullRefPtr<VFSRootContext> vfs_root_context, NonnullRefPtr<HostnameContext> hostname_context, RefPtr<Custody> current_directory, RefPtr<Custody> executable, RefPtr<TTY> tty, Process* fork_parent)
+ErrorOr<Process::ProcessAndFirstThread> Process::create_from_fork(Process& parent)
 {
+    VERIFY(!parent.is_kernel_process());
+    auto const& credentials = parent.credentials();
+
     Process::Name name {};
-    Process::current().name().with([&name](auto& process_name) {
+    parent.name().with([&name](auto& process_name) {
         name.store_characters(process_name.representable_view());
     });
-    return TRY(Process::create(name.representable_view(), uid, gid, ppid, is_kernel_process, move(vfs_root_context), move(hostname_context), current_directory, executable, tty, fork_parent));
+
+    return TRY(Process::create_impl(name.representable_view(),
+        credentials->uid(), credentials->gid(), parent.pid(),
+        parent.is_kernel_process(), parent.vfs_root_context(),
```

Comment:
This `parent.is_kerenl_process()` should be superfluous as you `VERIFY(!parent.is_kernel_process());` a couple lines above.

Reply:
You already pass `false` there for `Process::create_spawned()` :shruggie:

But I agree just having a plain bool there is not really nice.